docs/decisions/0004-toggle-api.rst

```diff
@@ -0,0 +1,26 @@
+Feature Toggle API
+==================
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+The setting toggles introduced in `ADR 0003 <../0003-django-setting-toggles>`__ will be added to this repository. In order to expose a consistent API, the public classes that should be imported from other repositories, such as ``edx-platform``, will be accessible as follows::
+
+    from edx_toggles.toggles import SettingToggle, SettingDictToggle
+
+Decision
+--------
+
+Objects that do not belong to the public API will be moved to an ``internal`` module. Public API objects will be exposed as follows in ``edx_toggles/toggles/__init__.py``::
+
+    from .internal import ...
+
+Consequences
+------------
+
+Whenever a new class or function is added to the edx_toggles public API, it should be implemented in the ``internal`` module and explicitly imported in the ``edx_toggles/toggles/__init__.py`` module.
```

Comment:
Thanks for this.  I was planning on a directory, rather than a module, because I imagine we want more modules.  Otherwise, an ``internal.py`` module would have been the simplest version of the idea.

```suggestion
Decision
--------

Objects that do not belong to the public API will be moved to an ``internal`` directory. Public API objects will be exposed as follows in ``edx_toggles/toggles/__init__.py``::

    from .internal.toggles import ...
    from .internal.waffle import ...

The benefits of this setup include:

* A clear designation of what is part of the public API.
* The ability to refactor the implementation without changing the API.
* A clear reminder to developers adding new code that it needs to be exposed if it is public.

Consequences
------------

Whenever a new class or function is added to the edx_toggles public API, it should be implemented in the ``internal`` directory and explicitly imported in the ``edx_toggles/toggles/__init__.py`` module.
```

Reply:
I purposefully used the word "module" and not "file" or "directory". This leaves us the possibility to split the `internal.py` file in multiple files later on. If you still feel like you prefer the wording with directories, I'll go with your change.

Reply:
1. The example import you provided made me think "file", but if you are happy with "module", go for it.  What do you think of the example imports I provided, which I think is going to be our likely usage?
2. Re-reading the sentence `Objects that do not belong to the public API will be moved to..`, I find it potentially misleading. It reads as-if public methods could live outside of `internal`, which would in turn just be part of the public API.  As an alternative, 2 separate statements:
```
All implementation code should be moved to an ``internal`` module.
The Public API will be exposed as follows in ``edx_toggles/toggles/__init__.py``
```
3. I'd like to keep the `benefits` section I added.

Reply:
:ok_hand: